subxt/src/client/online_client.rs

```diff
@@ -237,18 +195,51 @@ impl<T: Config> OnlineClient<T> {
         inner.metadata.clone()
     }
 
+    /// Change the [`Metadata`] used in this client.
+    ///
+    /// # Warning
+    ///
+    /// Setting custom metadata may leave Subxt unable to work with certain blocks,
+    /// subscribe to latest blocks or submit valid transactions.
+    pub fn set_metadata(&self, metadata: Metadata) {
+        let mut inner = self.inner.write();
+        inner.metadata = metadata;
+    }
+
     /// Return the genesis hash.
     pub fn genesis_hash(&self) -> T::Hash {
         let inner = self.inner.read();
         inner.genesis_hash
     }
 
+    /// Change the genesis hash used in this client.
+    ///
+    /// # Warning
+    ///
+    /// Setting a custom genesis hash may leave Subxt unable to
+    /// submit valid transactions.
+    pub fn set_genesis_hash(&self, genesis_hash: T::Hash) {
+        let mut inner = self.inner.write();
+        inner.genesis_hash = genesis_hash;
+    }
+
     /// Return the runtime version.
     pub fn runtime_version(&self) -> RuntimeVersion {
         let inner = self.inner.read();
         inner.runtime_version.clone()
     }
 
+    /// Change the [`RuntimeVersion`] used in this client.
+    ///
+    /// # Warning
+    ///
+    /// Setting a custom runtime version may leave Subxt unable to
+    /// submit valid transactions.
+    pub fn set_runtime_version(&self, runtime_version: RuntimeVersion) {
```

Comment:
Personally I would have the runtime_version and metadata setting in one method, rather than one each. Makes it less likely they are put out of sync, and also means we only need to acquire the lock once when we are updating both runtime the metadata (which we should be doing). But this would work

Reply:
I did contemplate this, as I'm def aware of the extra locking and synchronisation things, but from an API perspective it just felt nicer to keep them separate and ignore the fact that it happens to lock under the hood (and the locking should be very trivial anyway in terms of time cost), and for your sake you probably only really ever need to update the `Metadata` and can ignore the rest :)